velox/connectors/tpch/TpchConnector.cpp

```diff
@@ -127,21 +128,31 @@ void TpchDataSource::addSplit(std::shared_ptr<ConnectorSplit> split) {
       "Previous split has not been processed yet. Call next() to process the split.");
   currentSplit_ = std::dynamic_pointer_cast<TpchConnectorSplit>(split);
   VELOX_CHECK(currentSplit_, "Wrong type of split for TpchDataSource.");
-  splitOffset_ = 0;
+
+  size_t partSize =
+      std::ceil((double)tpchTableRowCount_ / (double)currentSplit_->totalParts);
```

Comment:
For simplicity, how about
```
std::ceil(tpchTableRowCount_ / currentSplit_->totalParts * 1.0);
```
or just one (double) conversion? The compiler guarantees the conversion on the second operand.